app/src/main/java/de/tum/in/tumcampusapp/component/tumui/grades/GradesFragment.kt

```diff
@@ -134,7 +137,11 @@ class GradesFragment : FragmentForAccessingTumOnline<ExamList>(
             setDrawEntryLabels(false)
             legend.isWordWrapEnabled = true
             description = null
-
+            // the legend should not contain all possible grades but rather the most common ones
+            legend.setEntries(legend.entries.filter {
+                it.label != null && !it.label.contains(Regex("[1-5],[1245689]"))
```

Comment:
We probably don't want to recompile the Regex for every entry we want to filter. Could you please move the regex object to a static / companion variable?

Reply:
Yup, thanks for the tip, doing that!